### DISCUSSION TITLE
What doesn't work with SWDCOM on RP2040?

### DISCUSSION THREAD

**Discussion by znmeb** [tabemann/zeptoforth#31] [Q&A] [answered]
The README states

"from release 0.35.0 on the rp2040 platform does not have the full_swdcom or mini_swdcom builds as swdcom does not properly function on this platform"

What is the nature of the failure? I'm not at all familiar with SWD yet, but I've hooked up a second Pico running `picoprobe` as shown in the SDK documentation and `gdb` works as advertised.

**Comment by tabemann** [comment, marked as answer]
The limitation is swdcom's, not zeptoforth - I just have stopped making swdcom builds for zeptoforth because swdcom cannot use them anyways. The specific issue is that swdcom relies on libstlink to talk to an ST-Link device which talks to a board, and libstlink lacks support for the RP2040 even though the RP2040 exposes SWD.

**Reply by znmeb** [reply]
Ah, OK. I'm getting some of the new Raspberry Pi Debug Probes so I'll be able to do both SWD and Serial-over-USB connections to the RP2040 boards. I have ST-LINK connectors for all my Daisy boards but I'm months away from wanting to run zeptoforth to them, if I ever do.
